modules/t/DumpWGAExpectedTags.t

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env perl
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+use strict;
+use warnings;
+
+use Test::More;
+
+use Bio::EnsEMBL::Test::MultiTestDB;
+use Bio::EnsEMBL::Hive::DBSQL::DBConnection;
+use Bio::EnsEMBL::Hive::Utils::Test qw(standaloneJob);
+
+use File::Temp qw/tempfile/;
+
+# Check module can be seen and compiled
+use_ok('Bio::EnsEMBL::Compara::RunnableDB::GeneTrees::DumpWGAExpectedTags');
+
+# Load test DB
+my $multi_db = Bio::EnsEMBL::Test::MultiTestDB->new( "test_master" );
+my $compara_dba = $multi_db->get_DBAdaptor( "compara" );
+my $dbc = Bio::EnsEMBL::Hive::DBSQL::DBConnection->new(-dbconn => $compara_dba->dbc);
+my $compara_db = $dbc->url;
+
+# Number of tags before calling the module
+my $sql1 = "SELECT COUNT(*) AS total_rows FROM method_link_species_set_tag";
+my $db_vals1 = $dbc->db_handle->selectrow_hashref($sql1);
+my $all_count = $db_vals1->{total_rows};
+
+my $sql2 = "SELECT COUNT(*) AS wga_exp_rows FROM method_link_species_set_tag WHERE tag = 'wga_expected'";
+my $db_vals2 = $dbc->db_handle->selectrow_hashref($sql2);
+my $we_count = $db_vals2->{wga_exp_rows};
+
+my ($wefh, $test_wga_expected_file) = tempfile();
+standaloneJob(
+    'Bio::EnsEMBL::Compara::RunnableDB::GeneTrees::DumpWGAExpectedTags', # module
+    { # input param hash
+        'wga_expected_file' => $test_wga_expected_file,
+        'compara_db'        => $compara_db,
+    }
+);
+
+# Check that output file exists and is correct
+my $exp_output = [
+    "method_link_species_set_id\twga_expected\n",
+    "101\t1\n",
+    "102\t0\n",
+    "104\t0\n",
+    "105\t1\n",
+    "106\t1\n",
+];
+
+open( my $fh, '<', $test_wga_expected_file) or die "Cannot open $test_wga_expected_file for reading\n";
+my @got_output = <$fh>;
```

Comment:
I'm not 100% sure, but cannot you use `$wga_exp_fh` here? The file handler is open to read but hasn't started reading, so it should work as expected.
Alternatively, you could use `Bio::EnsEMBL::Utils::IO::slurp()` to read the contents into a reference variable. It has some additional checks that can be useful.

Reply:
Honestly, I haven't tried. I was following the logic from [here](https://github.com/Ensembl/ensembl-compara/blob/master/modules/t/HomologyIDMapping.t#L47) (lines 47, 55, 69).